superset-frontend/src/components/Timer.tsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useEffect, useRef, useState } from 'react';
+import { Label } from 'react-bootstrap';
+
+import { now, fDuration } from '../modules/dates';
+
+interface TimerProps {
+  endTime?: number;
+  isRunning: boolean;
+  startTime?: number;
+  status?: string;
+  style?: object;
+}
+
+export default function Timer({
+  endTime,
+  isRunning,
+  startTime,
+  status = 'success',
+  style,
+}: TimerProps) {
+  const [clockStr, setClockStr] = useState<string>('');
+  const timer = useRef<NodeJS.Timeout>();
```

Comment:
any reason not to use state for this instead of a ref? The component should be rerendering super often anyway, so i don't think it should have perf implications

Reply:
I was waiting for someone to look at this from performance POV. Thanks for this. will change it to useState. I had put useRef considering the timer was to be set and unset very often. But useState makes it simpler.